### ISSUE TITLE
Why does the order have processing status when it's "pending_review"

### ISSUE THREAD

**Issue by zztimur** [reverbdotcom/reverb-magento#247]
Magento order status changes to processing, but the order is "pending_review" in reverb order status.
At which point is it safe to ship?

**Comment by kylecrum** [comment]
@zztimur it's safe to ship when it is paid . pending review means that our fraud team is looking at it.

**Comment by zztimur** [comment]
Wouldn't it be safer to have a status on magento side that would be "pending reverb" so we don't ship until you clear the transaction and move it to the next step - "processing"?

**Comment by kylecrum** [comment]
@zztimur to me, "processing" means that the payment is still in process. I hope that people don't ship goods before the payment has cleared.

**Comment by zztimur** [comment]
Magento order statuses are:
Pending: Pending orders are brand new orders that have not been processed. Typically, these orders need to be invoiced and shipped.
Pending PayPal: Pending PayPal orders are brand new orders that have not been cleared by PayPal. [...]
Processing: Processing means that orders have either been invoiced or shipped, but not both.
Complete: Orders marked as complete have been invoiced and have shipped.
Cancelled: Cancelled orders should be used if orders are cancelled or if the orders have not been paid for.
Closed: Closed orders are orders that have had a credit memo assigned to it and the customer has been refunded for their order.
On Hold: Orders placed on hold must be taken off hold before continuing any further actions.
We need to get a different status for reverb-review and reverb-passed-review. 

Makes sense?

**Comment by kylecrum** [comment]
@zztimur I don't like any of those, statuses, unfortunately :(

I mean, we can make it Pending PayPal, that's the most accurate, although, it is not a Paypal payment, but the same process is happening there.

Isn't there an on hold status?

**Comment by zztimur** [comment]
You can introduce new ones. Sean would know.
So instead of it’d be “Reverb Review” and then “Reverb Passed” or something along those lines.

**Comment by dunagan5887** [comment]
There is an On Hold Magento Order status. Basically in this status no actions will be allowed to be taken on the order until it is taken out of Hold state. 

In terms of creating new statuses for Magento Order objects, I would strongly advise against doing so. A lot of Magento functionality is based off of conditions regarding an order's status, and creating new statuses could lead to unintended consequences

**Comment by zztimur** [comment]
@dunagan5887, can reverb-review set order status as "on-hold",
and then once it passes review "processing"?

Right now the status is set to processing while on review.

**Comment by dunagan5887** [comment]
I'm going to need some catching up here. Are "Reverb Review" and "Reverb Passed" statuses that are communicated from Reverb to Magento?

**Comment by kylecrum** [comment]
no, those would be custom statuses, I think. I'd like to just use the built-in On Hold status if we can.

**Comment by dunagan5887** [comment]
I guess I need clarification on the following:

> can reverb-review set order status as "on-hold"

Do you mean put to Magento-status-On-Hold manually by an admin? 

> then once it passes review "processing"?

Currently once Reverb communicates the Reverb-Order-Status-"Paid" to Magento, the Magento Order is set to Magento-status-Processing by creating an invoice

> Right now the status is set to processing while on review

I'm confused as to how this is happening. Are you saying that the Reverb-Order-Status-"Paid" communication is being made to Magento even though Reverb orders haven't been paid? 

It may be easier to have a call regarding all of this

**Comment by kylecrum** [comment]
@dunagan5887 we have a status called "pending_review" and I think the question is what to translate that to in Magento terms.

**Comment by dunagan5887** [comment]
That sounds like "Hold" to me

**Comment by zztimur** [comment]
Yes, but currently it sets as "processing"

**Comment by dunagan5887** [comment]
> Yes, but currently it sets as "processing"

What is "it" that is setting it to Processing?

**Comment by kylecrum** [comment]
@zztimur please link to code that you feel may be incorrect.

**Comment by zztimur** [comment]
https://github.com/reverbdotcom/reverb-magento/blob/2b9f4b53e894c88c7d638f5a7bf1fa61a51192dc/app/code/community/Reverb/ReverbSync/Model/Sync/Order/Update.php#L40

**Comment by dunagan5887** [comment]
Currently the only Reverb-Order-Status values which trigger any Magento functionality are "cancelled" and "paid". If Reverb sends Reverb-Order-Status-"Pending Review" the module will not place the order in Processing state. There are currently no event observer listening for pending-review

**Comment by zztimur** [comment]
@kylecrum What's the sequence of statuses on reverb side?

**Comment by kylecrum** [comment]
@zztimur depends on the order. However, when it reaches "paid", then it's safe to ship. Order table here: https://dev.reverb.com/docs/syncing-orders

**Comment by kylecrum** [comment]
In general, "paid" is your magic state of saying "it's ok to ship".

**Comment by zztimur** [comment]
We’ve had an order with reverb_status: pending_review and magent: processing.
Is it possible that something was paid and then went under review?

**Comment by kylecrum** [comment]
@zztimur you'd have to give me details, but once it is paid, we will not put it under review again.

**Comment by dunagan5887** [comment]
> We’ve had an order with reverb_status: pending_review and magent: processing.

That would be possible if a Magento admin manually tried to trigger an invoice, or if Reverb sent a "pending_review" after sending a "paid" status
